odoo/addons/base/models/qweb.py

```diff
@@ -1063,13 +1063,43 @@ def _compile_directive_content(self, el, options):
         body = []
         if el.text is not None:
             body.append(self._append(ast.Str(pycompat.to_text(el.text))))
-        if el.getchildren():
-            for item in el:
-                # ignore comments & processing instructions
-                if isinstance(item, etree._Comment):
-                    continue
-                body.extend(self._compile_node(item, options))
-                body.extend(self._compile_tail(item))
+
+        # group children between "trivial" (no dynamic attributes, no namespaces
```

Comment:
@xmo-odoo 
Very interesting. In case you are interested, I have some (still preliminary) work on a new QWeb compiler for Owl 2.0. 

The main idea is that the compiling process is made in two phases: from XML to a "QWeb AST", and only then to a compiled function.  This allow the parser to detect some specific properties of the template nodes.  For example, static nodes (no directive, for the node and all its children), are detected and appear as a special node in the AST (https://github.com/odoo/owl/blob/master-experiment/src/qweb/types.ts#L37). 

Then, when the code is generated, Owl can build the static node in a cache, and generate a simple ID in the compiled template, which will then simply do a `cloneNode(deep)` when rendered (obviously, this specific part does not apply to python QWeb).

I also considered having a specific node for "mostly static nodes": a dom node with dynamic attributes or event handlers, but completely static children. I may do that, but I aim first to have a fully usable reimplementation of Owl.

But back to QWeb compilation. Doing it in two phases allow me to perform a lot of small optimisations in the compiled code.  For example, Owl 2.0 templates will make the distinction between two cases: if the children of a node are known at compile time, or if the number of children is unknown (when there is a `t-if`, or `t-foreach`). If the children are known at compile time, QWeb can generate better code (it creates a list in one pass, instead of creating an empty list, then pushing in it for every children).

Reply:
@ged-odoo nice! Did you do any performance tests? :D

Reply:
It's a looot faster in some early tests, but I can't give a number.
1. the feature set implemented is not the same => the comparison is not fair
2. the performance improvement depends on the shape of the template. The more static a template is, the most it is improved. So it is easy to have big improvement numbers, but it is not representative of a real template.

Reply:
@ged-odoo could be interesting for general-purpose qweb stuff, but here the goal is very specifically the assets issue which can get seriously degenerate e.g. the root of assets_backend can have literally hundreds of "trivial" children.